activerecord/lib/active_record/autosave_association.rb

```diff
@@ -417,7 +417,8 @@ def save_has_one_association(reflection)
                 record[reflection.foreign_key] = key
               end
 
-              saved = record.save(:validate => !autosave)
+              _validate = !autosave && validate
```

Comment:
use: `validate = !autosave && self.validate`. BTW, where this `validate` is defined?